isb_web/analytics.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import typing
+import requests
+import json
+import isb_web.config
+from isb_format import _NoValue
+from isb_lib.core import MEDIA_JSON
+
+ANALYTICS_URL = isb_web.config.Settings().analytics_url
+ANALYTICS_DOMAIN = isb_web.config.Settings().analytics_domain
+
+
+class AnalyticsEvent(_NoValue):
+    """Enum class representing all possible analytics events we may record.  These need to be defined in plausible.io
+    per https://plausible.io/docs/custom-event-goals#using-custom-props"""
+
+    THING_LIST = "thing_list"
+    THING_LIST_METADATA = "thing_list_metadata"
+    THING_LIST_TYPES = "thing_list_types"
+    THING_SOLR_SELECT = "thing_solr_select"
+    THING_SOLR_LUKE_INFO = "thing_solr_luke_info"
+    THING_SOLR_STREAM = "thing_solr_stream"
+    THING_BY_IDENTIFIER = "thing_by_identifier"
+    STAC_ITEM_BY_IDENTIFIER = "stac_item_by_identifier"
+    STAC_COLLECTION = "stac_collection"
+    RELATED_METADATA = "related_metadata"
+    RELATED_SOLR = "related_solr"
```

Comment:
In theory we have one event per API call.  To get these to show up in the dashboard, they also need to be added as conversion goals in the plausible.io admin UI.